src/librustc_interface/util.rs

```diff
@@ -109,9 +109,6 @@ pub fn get_codegen_backend(sess: &Session) -> Box<dyn CodegenBackend> {
         let codegen_name = sess.opts.debugging_opts.codegen_backend.as_ref()
             .unwrap_or(&sess.target.target.options.codegen_backend);
         let backend = match &codegen_name[..] {
-            "metadata_only" => {
```

Comment:
Since we're removing this here, I assume we also need to remove this below?
https://github.com/rust-lang/rust/blob/350674b7180a41c8e508d93c6ab8e203b69d3df7/src/librustc_driver/test.rs#L100


This was introduced recently in https://github.com/rust-lang/rust/commit/23a51f91c928a4ff2cbf39218e6e991365e5f562
cc @Zoxc

Reply:
The `rustc_driver` tests rely on the metadata-only backend since it can't find the real one. How is that resolved in this PR?

Reply:
Tests in `rustc_driver` is an artifact of days long before 1.0 (just look at what they are doing - testing lifetime inference? in rustc_driver?).
They should be just removed at long last.

Reply:
Now I get why it failed testing. I thought it was commit 72dffcc, which I removed after seeing the failing test. I will remove the tests from rustc_driver and reintroduce 72dffcc.